src/Controller/Entry/EntryFrontController.php

```diff
@@ -5,213 +5,132 @@
 namespace App\Controller\Entry;
 
 use App\Controller\AbstractController;
-use App\Controller\User\ThemeSettingsController;
 use App\Entity\Magazine;
 use App\Entity\User;
 use App\PageView\EntryPageView;
+use App\PageView\PostPageView;
 use App\Pagination\Pagerfanta as MbinPagerfanta;
-use App\Repository\Criteria;
 use App\Repository\EntryRepository;
+use App\Repository\PostRepository;
 use Pagerfanta\PagerfantaInterface;
 use Symfony\Bridge\Doctrine\Attribute\MapEntity;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\Security\Http\Attribute\IsGranted;
 
 class EntryFrontController extends AbstractController
 {
-    public function __construct(private readonly EntryRepository $repository)
+    public function __construct(private readonly EntryRepository $entryRepository, private readonly PostRepository $postRepository)
     {
     }
 
-    public function root(?string $sortBy, ?string $time, ?string $type, Request $request): Response
+    public function root(?string $sortBy, ?string $time, ?string $type, string $federation, Request $request): Response
     {
-        $user = $this->getUser();
-
-        if (!$user) {
-            return $this->front($sortBy, $time, $type, $request);
-        }
-
-        $front = match ($user->homepage) {
-            User::HOMEPAGE_SUB => 'subscribed',
-            User::HOMEPAGE_MOD => 'moderated',
-            User::HOMEPAGE_FAV => 'favourite',
-            default => 'front',
-        };
-
-        return $this->$front($sortBy, $time, $type, $request);
+        return $this->front($sortBy, $time, $type, $request->query->get('subscription'), $federation, $request);
```

Comment:
`content` is missing in the method declaration

Reply:
It was dead code because we didn't use the 'root' route; I eliminated ::root() and the root route completely.